doc/string.rdoc

```diff
@@ -47,4 +47,81 @@ class String
   def initialize(str = '', encoding: nil, capacity: nil)
     Primitive.rb_str_init(str, encoding, capacity)
   end
+
+  #  call-seq:
+  #    split(pattern = $FIELD_SEPARATOR, limit = nil) -> array
+  #    split(pattern = $FIELD_SEPARATOR, limit = nil) {|sub| block } -> object
+  #
+  #  Returns an array of substrings of +self+;
+  #  the substrings are copied from +self+, based on the given arguments.
```

Comment:
I think the original description was more clear:

> Divides `self` into substrings based on a delimiter, returning an array
> of these substrings.

Reply:
I'll think on this; I thought "Divides self" as the first two words suggests that self is modified.

Reply:
Yeah perhaps the original description isn't great either. But I think there should be a mention of "splitting" or "dividing" the original string.

Reply:
Reworded.